internal/provider/resource_deployment.go

```diff
@@ -831,6 +873,15 @@ func (d *DeploymentDeployer) updateState(ctx context.Context, sub *substrate.Sub
 				zdbNamespace[string(w.Name)] = d.Namespace
 			} else if w.Type == zos.QuantumSafeFSType {
 				workloads[string(w.Name)] = &dl.Workloads[idx]
+			} else if w.Type == zos.ZLogsType {
+				d, err := w.WorkloadData()
+				if err != nil {
+					log.Printf("error loading machine data: %s\n", err)
+					continue
+				}
+				zmachine := d.(*zos.ZLogs).ZMachine.String()
+				output := d.(*zos.ZLogs).Output
+				vmZlog[zmachine] = append(vmZlog[zmachine], Zlog{output})
```

Comment:
If the node returns the workloads out of order, it will cause the local state to be updated with a different order than it has, incorrectly detecting a change? And then when a user does an apply it will log that the local state doesn't match the remote and needs an action